commons/src/main/kotlin/com/simplemobiletools/commons/dialogs/ChangeDateTimeFormatDialog.kt

```diff
@@ -71,3 +95,196 @@ class ChangeDateTimeFormatDialog(val activity: Activity, val callback: () -> Uni
         return DateFormat.format(format, cal).toString()
     }
 }
+
+
+@Composable
+fun ChangeDateTimeFormatAlertDialog(
+    alertDialogState: AlertDialogState,
+    modifier: Modifier = Modifier,
+    is24HourChecked: Boolean,
+    callback: (selectedFormat: String, is24HourChecked: Boolean) -> Unit
+) {
+    val context = LocalContext.current
+    val selections = remember {
+        mapOf(
+            Pair(DATE_FORMAT_ONE, formatDateSample(DATE_FORMAT_ONE)),
+            Pair(DATE_FORMAT_TWO, formatDateSample(DATE_FORMAT_TWO)),
+            Pair(DATE_FORMAT_THREE, formatDateSample(DATE_FORMAT_THREE)),
+            Pair(DATE_FORMAT_FOUR, formatDateSample(DATE_FORMAT_FOUR)),
+            Pair(DATE_FORMAT_FIVE, formatDateSample(DATE_FORMAT_FIVE)),
+            Pair(DATE_FORMAT_SIX, formatDateSample(DATE_FORMAT_SIX)),
+            Pair(DATE_FORMAT_SEVEN, formatDateSample(DATE_FORMAT_SEVEN)),
+            Pair(DATE_FORMAT_EIGHT, formatDateSample(DATE_FORMAT_EIGHT)),
+        )
+    }
+    val kinds = remember {
+        selections.values.toImmutableList()
+    }
+    val initiallySelected = remember {
+        requireNotNull(selections[context.baseConfig.dateFormat]){
+            "Incorrect format, please check selections"
+        }
+    }
+    AlertDialog(
+        onDismissRequest = alertDialogState::hide,
+    ) {
+        Column(
+            modifier = modifier
+                .fillMaxWidth()
+                .background(dialogContainerColor, dialogShape)
+                .verticalScroll(rememberScrollState())
+                .padding(16.dp)
+        ) {
+            var is24HoursSelected by remember { mutableStateOf(is24HourChecked) }
+
+            val (selected, setSelected) = remember { mutableStateOf(initiallySelected) }
+            RadioGroup(
+                items = kinds, selected = selected,
+                setSelected = setSelected,
+                modifier = Modifier.padding(
+                    vertical = 16.dp,
+                    horizontal = 11.dp
```

Comment:
lets use radiobuttons that are reusable in other dialogs and apps too, do not hardcode values here